modules/json-schema/src/internals/Extractors.scala

```diff
@@ -183,6 +183,12 @@ object Extractors {
         val genericHints = getGenericHints(sch)
         Option(obj.getPatternProperties().asScala.head._2)
           .map(genericHints -> _)
+      case (obj: ObjectSchema)
+          if obj.getPropertySchemas().size() == 0 && obj
+            .getSchemaOfAdditionalProperties() != null =>
+        val genericHints = getGenericHints(sch)
+        Option(obj.getSchemaOfAdditionalProperties())
```

Comment:
you probably don't have to wrap into `Option` since you're already checking for null